From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: mask raw in struct bpf_reg_state
Patch-mainline: No, SUSE-specific
References: bsc#1083647

patches.fixes/bpf-fix-partial-copy-of-map_ptr-when-dst-is-scalar.patch
introduced a new member 'unsigned long raw' to the union member of
'bpf_reg_state'. Since there is already a 'struct bpf_map *' member in
the union, it's safe to append 'raw' without breaking kABI.

Signed-off-by: Gary Lin <glin@suse.com> 

---
 include/linux/bpf_verifier.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -51,8 +51,10 @@ struct bpf_reg_state {
 		 */
 		struct bpf_map *map_ptr;
 
+#ifndef __GENKSYMS__
 		/* Max size from any of the above. */
 		unsigned long raw;
+#endif
 	};
 	/* Fixed part of pointer offset, pointer types only */
 	s32 off;
